File: app/src/main/java/za/co/gundula/app/arereyeng/model/ApiToken.java
package za.co.gundula.app.arereyeng.model;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by kgundula on 2016/11/09.
 */

public class ApiToken implements Parcelable {

    private String access_token;
    private String expires_in;
    private String token_type;

    public ApiToken() {
    }

    public ApiToken(String access_token, String expires_in, String token_type) {
        this.access_token = access_token;
        this.expires_in = expires_in;
        this.token_type = token_type;
    }

    protected ApiToken(Parcel in) {
        access_token = in.readString();
        expires_in = in.readString();
        token_type = in.readString();
    }

    public static final Creator<ApiToken> CREATOR = new Creator<ApiToken>() {
        @Override
        public ApiToken createFromParcel(Parcel in) {
            return new ApiToken(in);
        }

        @Override
        public ApiToken[] newArray(int size) {
            return new ApiToken[size];
        }
    };

    public String getAccess_token() {
        return access_token;
    }

    public void setAccess_token(String access_token) {
        this.access_token = access_token;
    }

    public String getExpires_in() {
        return expires_in;
    }

    public void setExpires_in(String expires_in) {
        this.expires_in = expires_in;
    }

    public String getToken_type() {
        return token_type;
    }

    public void setToken_type(String token_type) {
        this.token_type = token_type;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(access_token);
        parcel.writeString(expires_in);
        parcel.writeString(token_type);
    }
}

File: app/src/main/java/za/co/gundula/app/arereyeng/model/ApiToken.kt
package za.co.gundula.app.arereyeng.model

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by kgundula on 2016/11/09.
 */
class ApiToken : Parcelable {
    var accessToken: String? = null
    var expiresIn: String? = null
    var tokenType: String? = null

    constructor(access_token: String?, expires_in: String?, token_type: String?) {
        this.accessToken = access_token
        this.expiresIn = expires_in
        this.tokenType = token_type
    }

    protected constructor(`in`: Parcel) {
        accessToken = `in`.readString()
        expiresIn = `in`.readString()
        tokenType = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeString(accessToken)
        parcel.writeString(expiresIn)
        parcel.writeString(tokenType)
    }

    companion object CREATOR : Parcelable.Creator<ApiToken> {
        override fun createFromParcel(parcel: Parcel): ApiToken {
            return ApiToken(parcel)
        }

        override fun newArray(size: Int): Array<ApiToken?> {
            return arrayOfNulls(size)
        }
    }

}